### PR TITLE
add bad password tests to TDVT

### PR DESCRIPTION
**Pull request by sonia0504**
add pretest folder with one test config file and expected file
add function get_expected_message
modify tdvt.py, tdvt_core.y, test_result.py, tabquery.py, test_config.py and datasource_list.py to recognize "expected_message" so that test failures for bad password test will be reported as 'passed' if actual errors contain "expected_message".

### CODE REVIEW

**Comment by lukewrites** [comment]
Looks like we need to bump the version number in `version.py`?

File: tdvt/tdvt/tdvt_core.py [hunk 1]
```diff
@@ -475,6 +490,8 @@ def get_csv_row_data(tds_name, test_name, test_path, test_result, test_case_inde
     if not test_result or not test_result.get_test_case_count() or not test_result.get_test_case(test_case_index):
         error_msg= test_result.get_failure_message() if test_result else None
         error_type= test_result.get_failure_message() if test_result else None
+        if isinstance(test_result.error_status, TestErrorExpected):
```
Comment (lriggs) [inline tdvt/tdvt/tdvt_core.py]: I think this should go into the TestResult class in test_results.py. You could probably update all_passed there and use that.

File: tdvt/tdvt/test_results.py [hunk 2]
```diff
@@ -223,7 +236,9 @@ def get_name(self):
 
     def all_passed(self):
         """Return true if all aspects of the test passed."""
-        if self.error_status or not self.test_case_map:
+        if isinstance(self.error_status, TestErrorExpected):
```
Comment (lriggs) [inline tdvt/tdvt/test_results.py]: Can you make a helper method for all these isinstance calls?

File: tdvt/tdvt/tdvt_core.py [hunk 3]
```diff
@@ -545,7 +562,8 @@ def write_csv_test_output(all_test_results, tds_file, skip_header, output_dir):
         test_name = test_result.get_name() if test_result.get_name() else path
         if not test_result or not test_result.get_test_case_count():
             csv_out.writerow(get_csv_row_data(tdsname, test_name, path, test_result))
-            total_failed_tests += 1
+            if not isinstance(test_result.error_status, TestErrorExpected):
```
Comment (lriggs) [inline tdvt/tdvt/tdvt_core.py]: Should go into test_results.

File: tdvt/tdvt/config_gen/datasource_list.py [hunk 4]
```diff
@@ -289,7 +312,7 @@ def __init__(self, ini_file):
         self.load_ini_file(ini_file)
 
     def load_ini_file(self, ini_file):
-        #Create the test suites (groups of datasources to test)
+        # Create the test suites (groups of datasources to test)
```
[the remark this answers is not in the document]
Comment (sonia0504) [reply tdvt/tdvt/config_gen/datasource_list.py]: yes, it recommended by pycharm per PEP 8

**Comment by lriggs** [comment]
Looks good

**Pull request merged**